src/main/java/org/wso2/carbon/mongodb/query/MongoPreparedStatementImpl.java

```diff
@@ -0,0 +1,707 @@
+        for (String elementName : elementNames) {
+            Object value = stmt.get(elementName);
+            if (parameterValue.containsKey(elementName)) {
+                Object val = parameterValue.get(elementName);
+                if (!(value instanceof JSONObject) && !val.equals("%")) {
+                    mapMatch.put(elementName, val);
+                } else {
+                    if (value instanceof JSONObject) {
+                        JSONObject match = (JSONObject) value;
+                        String[] elements = JSONObject.getNames(match);
+                        for (String element : elements) {
+                            if (!val.equals("%")) {
+                                if (element.equals(MONGO_REGEX_FIELD)) {
+                                    mapMatchCaseInSensitive.put(element, val);
+                                } else {
+                                    mapMatchCaseInSensitive.put(element, MONGO_CASE_INSENSITIVE_OPTION);
+                                }
+                                this.isCaseSensitive = false;
+                            }
+                        }
+                    } else {
+                        if (!val.equals("%")) {
+                            mapMatch.put(elementName, val);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * JSON query will update with  user values
+     *
+     * @param stmt to update
+     */
+    private void setUpdateObject(JSONObject stmt) {
+        String[] elementNames = JSONObject.getNames(stmt);
+        for (String elementName : elementNames) {
+            if (parameterValue.containsKey(elementName)) {
+                Object val = parameterValue.get(elementName);
+                mapProjection.put(elementName, val);
+            }
+        }
+        if (!mapProjection.isEmpty()) {
+            this.projection = new BasicDBObject(mapProjection);
+        }
+    }
+
+    /**
+     * Check for multiple look up for aggregation pipeline
+     *
+     * @return boolean status
+     */
+    private boolean isMultipleLookUp() {
+        return multipleLookUpStatus;
+    }
+
+    /**
+     * Add lookup attribute to pipeline
+     *
+     * @param pipeline pipeline list, which needs to be modified
+     */
+    private void addLookUpAttribute(List<DBObject> pipeline) {
+        if (isMultipleLookUp()) {
+            DBObject lookup = new BasicDBObject("$lookup", new BasicDBObject(mapLookUp));
+            pipeline.add(lookup);
+        } else {
+            int track = 0;
+            // Add the json query object to aggregation pipeline in order manner
+            while (track < multiMapLookup.size()) {
+                for (Map<String, Object> map : multiMapLookup) {
+                    if (map.containsKey("dependency")) {
+                        map.remove("dependency");
+                        DBObject lookup = new BasicDBObject("$lookup", new BasicDBObject(map));
```

Comment:
Can we move the duplicated string values to constants

Reply:
Added constants to denote the recurring strings.